python/ql/src/semmle/python/frameworks/Django.qll

```diff
@@ -1872,7 +1872,22 @@ private module Django {
   private class DjangoUrlsRePathCall extends DjangoRegexRouteSetup {
     override CallNode node;
 
-    DjangoUrlsRePathCall() { node.getFunction() = django::urls::re_path().asCfgNode() }
+    DjangoUrlsRePathCall() {
+      node.getFunction() = django::urls::re_path().asCfgNode() and
+      // `django.conf.urls.url` (which we support directly with
+      // `DjangoConfUrlsUrlCall`), is implemented in Django 2+ as backward compatibility
+      // using `django.urls.re_path`. See
+      // https://github.com/django/django/blob/stable/3.2.x/django/conf/urls/__init__.py#L22
+      // Since we're still installing dependencies and analyzing their source code,
+      // without explicitly filtering out this call, we would be double-counting such
+      // route-setups :( One practical negative side effect of double-counting it, is
+      // that since we can't figure out the URL, we mark ANY parameter as being a
+      // routed-parameter, which can lead to FPs.
```

Comment:
I do not understand this. As far as I can tell they both (`DjangoUrlsRePathCall ` and `DjangoConfUrlsUrlCall `) extend `DjangoRegexRouteSetup` and neither override `getARoutedParameter`, so there should be no difference in results?

Reply:
They both provide implementations of `getUrlPatternArg`, and `DjangoRegexRouteSetup` provides a shared implementation of `getAROutedParameter` based on the result from `getUrlPatternArg`:

https://github.com/github/codeql/blob/8e5557eca37d9f28c4cddadca8d4a0478eaa5984/python/ql/src/semmle/python/frameworks/Django.qll#L1825-L1848

Reply:
Right, but they have equally specific implementations of `getUrlPatternArg` (index 0, named "route" in one and "regex" in the other), so I do not see a loss of precision?

Reply:
> since we can't figure out the URL [for the `DjangoUrlsRePathCall`], we mark ANY parameter as being a
> routed-parameter, which can lead to FPs.

Reply:
Ok, I now read this as "the condition on line 1830 distinguishes the two classes". Reading the definition of `getUrlPattern`, it seems to depend on flow being found into a result of `getUrlPatternArg`. Since the two classes have virtual identical implementations of this predicate, the difference would seem to arise from how code is written in the wild? Why is it that we cannot figure out the URL in one and can in the other?

Reply:
Thanks for the meeting, clarification suggested.